nixos/tests/kafka/cluster.nix

```diff
@@ -0,0 +1,170 @@
+import ../make-test-python.nix ({ lib, pkgs, ... }:
+
+let
+  inherit (lib) mkMerge;
+
+  kafkaConfig = {
+    networking.firewall.allowedTCPPorts = [ 9092 9093 ];
+
+    virtualisation.memorySize = 1024 * 2;
+
+    environment.systemPackages = [ pkgs.apacheKafka ];
+
+    services.apache-kafka = {
+      enable = true;
+
+      # Randomly generated uuid. You can get one by running:
+      # kafka-storage.sh random-uuid
+      clusterId = "pHG8aWuXSfWAibHFDCnsCQ";
+
+      formatLogDirs = true;
+
+      settings = {
+        listeners = [
+          "PLAINTEXT://:9092"
+          "CONTROLLER://:9093"
+        ];
+        "listener.security.protocol.map" = [
+          "PLAINTEXT:PLAINTEXT"
+          "CONTROLLER:PLAINTEXT"
+        ];
+        "controller.quorum.voters" = [
+          "1@kafka1:9093"
+          "2@kafka2:9093"
+          "3@kafka3:9093"
+          "4@kafka4:9093"
+        ];
+        "controller.listener.names" = ["CONTROLLER"];
+
+        "process.roles" = ["broker" "controller"];
+
+        "log.dirs" = ["/var/lib/apache-kafka"];
+        "num.partitions" = 6;
+        "offsets.topic.replication.factor" = 2;
+        "transaction.state.log.replication.factor" = 2;
+        "transaction.state.log.min.isr" = 2;
+      };
+    };
+
+    systemd.services.apache-kafka.serviceConfig.StateDirectory = "apache-kafka";
+  };
+
+  extraKafkaConfig = {
+    kafka1 = {
+      services.apache-kafka.settings = {
+        "node.id" = 1;
+        "broker.rack" = 1;
+      };
+    };
+
+    kafka2 = {
+      services.apache-kafka.settings = {
+        "node.id" = 2;
+        "broker.rack" = 2;
+      };
+    };
+
+    kafka3 = {
+      services.apache-kafka.settings = {
+        "node.id" = 3;
+        "broker.rack" = 3;
+      };
+    };
+
+    kafka4 = {
+      services.apache-kafka.settings = {
+        "node.id" = 4;
+        "broker.rack" = 3;
+      };
+    };
+  };
+
+  kafkaNodes = builtins.mapAttrs (_: val: mkMerge [ val kafkaConfig ]) extraKafkaConfig;
+in
+{
+  name = "kafka-cluster";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ nequissimus ];
```

Comment:
I don't like assigning new tests to someone else, could you set yourself as the maintainer of these new test?
CC @NeQuissimus in case you're also interested